### PR TITLE
TEST-#2755: add dtypes benchmark

### LINKED ISSUE #2755: [ASV] add dtypes benchmark

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [x] Resolves #2755 
- [ ] tests added and passing

### CODE REVIEW

File: asv_bench/benchmarks/benchmarks.py [hunk 1]
```diff
@@ -557,3 +557,8 @@ def time_columns(self, shape):
 
     def time_index(self, shape):
         return self.df.index
+
+    def time_dtypes(self, shape):
+        # trigger _compute_dtypes func
+        self.df._query_compiler._modin_frame._dtypes = None
+        return self.df.dtypes
```
Comment (dchigarev) [inline asv_bench/benchmarks/benchmarks.py:564]: This makes bench strictly engine specific, can we somehow allow it to execute only for pandas backend?

Comment (anmyachev) [reply asv_bench/benchmarks/benchmarks.py]: I think now it was enough to fix pandas case

Comment (dchigarev) [reply asv_bench/benchmarks/benchmarks.py]: @devin-petersohn do you see a more general way to time this?

Comment (devin-petersohn) [reply asv_bench/benchmarks/benchmarks.py]: This is useful to benchmark, maybe we should have a utility that wipes metadata caches?

Comment (anmyachev) [reply asv_bench/benchmarks/benchmarks.py]: Create separate tracker for this https://github.com/modin-project/modin/issues/2780

There is no blocker for dtypes benchmark I suppose.

**Pull request closed without merging**